### DISCUSSION TITLE
Why am I seeing jobs unexpectedly appear in the scheduled queue?

### DISCUSSION THREAD

**Discussion by dhughes** [sidekiq/sidekiq#6548] [Q&A]
This is probably a "me" problem, but I'm struggling to figure it out. A project I'm working on has historically used `perform_in` to schedule jobs, so as to space them out to try to avoid hogging resources. I'm switching that to simply use `perform_async` and enqueuing the jobs to a constrained queue in a capsule. The problem I'm having is that I'm still seeing jobs unexpectedly appearing in the scheduled queue. They're all scheduled for "right now". 

I've gone through my code with a fine toothed comb and I can't find any use of `perform_in` or `perform_at`. Is there some expected mechanism in Sidekiq by which jobs run with `perform_async` might end up in the scheduled queue? I'm using `set` to specify the queue for the job. EG: `SomeJob.set(queue: 'bulk_queue').perform_async(...)`. Maybe that's impacting things?

**Comment by dhughes** [comment]
Just to add a little bit more information to my question, this system runs workers in batches. Also, the workers employ rate limiters. I didn't notice any overlimit errors, so I don't think that's at all relevant.

**Reply by kkohrt** [reply]
I'm not sure I have a grasp on all the queue names in play, but I am curious what you mean when you said:
> this system runs workers in batches
Is that like an autoscaling thing where a worker is a server?
Or do you mean a Sidekiq batch job ("worker" means "job")?

**Reply by dhughes** [reply]
Worker means job, sorry. It's been ages and my mind is still stuck on the term "worker" when I mean job.

**Comment by mperham** [comment]
Any job with an `at` attribute will also be scheduled. It's unlikely but possible: `Sidekiq::Client.push("class" ..., "args" ..., "at" ...)`

**Comment by patrick-boatner** [comment]
Are you using https://github.com/sidekiq/sidekiq/wiki/Ent-Rate-Limiting? If you exceed an enterprise rate limit, which raises `Sidekiq::Limiter::OverLimit`, your job will be rescheduled.

Try grabbing the JID for an example job, and searching your logs to see what enqueued it.
